design/aws-ecs-scheduled-ecs-task-construct.md

```diff
@@ -0,0 +1,141 @@
+# AWS ECS - L3 Construct for Scheduling EC2 Tasks
+
+To address issue [#2352](https://github.com/awslabs/aws-cdk/issues/2352), the ECS CDK construct library should provide a way for customers to create a standalone scheduled task, without creating a service, for their container. The task will be initiated by a cloudwatch event that is scheduled based on a [CW Schedule Event](http://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html).
+
+This would mean adding a new ECS CDK construct `ScheduledEc2Task` and properties `ScheduledEc2TaskProps`, that would take in the necessary properties required to create a Task Definition, an EventRuleTarget as well as an AWS EventRule. To allow for flexibility, container overrides will also be supported properties.
+
+Note: Currently, there is only [CloudFormation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-reference-ecs.html) support for scheduled tasks on EC2, and not on Fargate. CloudFormation currently only supports clusters, services and taskdefinitions: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-reference-ecs.html.
+
+TaskDefinitions contain a network mode, and for EC2, we can use none, bridge or the many others. To support Fargate, we would need to use a awsvpc as the NetworkMode of the TaskDefinition, but this requires us to create a NetworkConfiguration, which is not supported in the TaskDefinition, but is supported in the Service.
```

Comment:
> NetworkConfiguration, which is not supported in the TaskDefinition, but is supported in the Service.

It would be more correct to say that it is *supported in the API*. Since both Services and Tasks can be created with a network configuration.

Reply:
Suggested rephrasing of the 2nd sentence, per the above:

"However, TaskDefinitions running on Fargate require the awsvpc network mode and additional NetworkConfiguration settings at the [service](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_CreateService.html) or [task](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_RunTask.html) level when deployed."